### PR TITLE
5 tests failed

### PR DESCRIPTION
**Pull request by outintotheblue**

### CODE REVIEW

File: week1/homework/src/server.js [hunk 1]
```diff
@@ -9,12 +9,33 @@ function createServer(port) {
   let state = 10;
 
   const server = http.createServer((request, response) => {
-    // TODO: Write your homework code here
+    response.writeHead(200, { 'Content-Type': 'application/json' });
```
Comment (apooslr) [inline week1/homework/src/server.js]: I would say that we should not send a response status code of 200 until we know what we have hit a valid path.

File: week1/homework/src/server.js [hunk 2]
```diff
@@ -9,12 +9,33 @@ function createServer(port) {
...
+    }
+    response.end();
   });
 
   return server;
 }
 
 module.exports = {
-  createServer
+  createServer,
```
Comment (marianabocoi) [inline week1/homework/src/server.js]: please remove comma :)

File: week1/homework/src/server.js [hunk 3]
```diff
@@ -9,12 +9,36 @@ function createServer(port) {
...
+        response.writeHead(200, { 'Content-Type': 'application/json' });
+        response.write(JSON.stringify({ state: state }));
+        break;
+      case '/subtract':
+        state -= 1;
+        response.writeHead(200, { 'Content-Type': 'application/json' });
+        response.write(JSON.stringify({ state: state }));
+        break;
+      case '/reset':
+        state == 10;
```
Comment (marianabocoi) [inline week1/homework/src/server.js]: what does this line (28) do?

**Review by marianabocoi — COMMENTED** [review]
Great!

One more thought, how can we have less duplication in the code?

File: week1/homework/src/server.js [hunk 4]
```diff
@@ -5,16 +5,40 @@ const http = require('http');
 /* `createServer` MUST return an instance of `http.Server` otherwise the tests
  * will fail.
  */
+
 function createServer(port) {
   let state = 10;
 
   const server = http.createServer((request, response) => {
-    // TODO: Write your homework code here
+    function writeLines() {
```
Comment (marianabocoi) [inline week1/homework/src/server.js:13]: This function should probably live outside the createServer block.

**Review by marianabocoi — APPROVED** [review]

**Pull request merged**